pkg/graveler/graveler.go

```diff
@@ -775,6 +775,7 @@ type StagingManager interface {
 	Set(ctx context.Context, st StagingToken, key Key, value *Value, overwrite bool) error
 
 	// Update updates a (possibly nil) value under the given staging token and key.
+	// Skip update in case value returned from 'updateFunc' is the same as the value passed to call.
```

Comment:
```suggestion
	// It skips update when the value returned from 'updateFunc' is the same as the value passed to call.
```
Also please clarify what "value" means here.